src/mon/MonmapMonitor.cc

```diff
@@ -146,9 +148,9 @@ void MonmapMonitor::apply_mon_features(const mon_feature_t& features)
     (pending_map.persistent_features ^
      (features & ceph::features::mon::get_persistent()));
 
-  if (new_features.empty()) {
-    dout(10) << __func__ << " features match current pending: "
-             << features << dendl;
+  if (new_features.empty() && pending_map.min_mon_release == min_mon_release) {
```

Comment:
i'd suggest move the check of `pending_map.min_mon_release == min_mon_release` down before setting `pending_map.min_mon_release`. and 1) do not set `new_features` if it is empty,  2) do not update `pending_map.persistent_features` if it is empty. or better off guarding all of them with `if (new_features.empty()`. and set a flag `changed` to true if either `features` or `min_mon_release` is updated in the `pending_map`

Reply:
cleaned up this function significantly.. take a look